### DISCUSSION TITLE
Idee carine per il Registro Elettronico

### DISCUSSION THREAD

**Discussion by giadaschiavone** [iisgiua/giuaschool#501] [Idee e proposte]
Ciao a tutti! Stavo pensando ha delle idee carine da inserire sul Registro Elettronico:

- firma in compresenza delle lezioni scolastiche;
- possibilità di inserimento di allegati e video nella lezione, compiti e annotazioni;
- inserimento del campo _visibile al solo docente_ nelle lezioni e nei compiti, per inserire note personali della lezione;
- aggiunta della funzione _nota didattica_ e _altro tipo di nota,_ oltre alla nota disciplinare, per evitare confusione;
- possibilità di inserire la tipologia di lezione (es: _lezione ordinaria_, _lezione laboratoriale_ etc.)
- possibilità di inserire direttamente i compiti alla firma della lezione;
- aggiunta della funzione calendario, dove si inseriscono eventi e attività di classe e personali;
- nella funzione _documenti_, aggiungere la funzione _documenti per la classe, documenti per il docente, documenti per la classe e materia, documenti per alunno, documenti personali_;
- aggiunta della funzione _condivisione materiale_, in cui il genitore o studente può inviare al docente il materiale didattico o disciplinare (viceversa, il docente può inviarlo al genitore o studente;
- aggiunta della funzione _elaborato_, dove si inseriscono gli elaborati degli esami, tesine o variabili;
- nello scrutinio, aggiungere la funzione di importazione da proposta a scrutinio, funzione dei verbali e variabili;
- aggiungere la funzione della modulistica;
- aggiungere la funzione di adozione libri;
- aggiungere la funzione dei pagamenti;
- aggiungere la funzione questionari;
- aggiunta della funzione di alternanza alla scuola e al lavoro;
- aggiunta della funzione dell'amministrazione digitale e dell'albo online;
- aggiunta della funzione della costruzione del sito web scolastico;
- aggiunta della funzione di posta e messaggi (sms);
- aggiungere la funzione comunicazione tramite posta, personalizzabili, con funzione di inviarli a chi vuoi (es. comunicazione uno ad alunni, comunicazione due a genitori e docenti);
- aggiunta della funzione esami di stato e di maturità;
- aggiunta della funzione delle firme digitali;
- aggiunta della funzione SIDI;
- modifica della carta intestata (ognuno può modificare la carta intestata del proprio istituto, decidendo di togliere _istituto d'istruzione superiore_, che di solito appare nella consegna delle credenziali);
- aggiunta della funzione di creazione dei documenti (come circolari) tramite il registro, con carta intestata, come per esempio Word, Open Office o Libre Office.

Aggiungere queste funzioni sarebbe un vantaggio per tutti, per facilitare il lavoro dei docenti e quello della segreteria.
Grazie di avermi ascoltato!

**Comment by trinko** [comment]
**Risposta breve:** diverse delle funzioni sono già presenti, altre non riguardano il registro elettronico (incentrato sul lavoro dei docenti e sui rapporti con la famiglia) ma il gestionale che deve gestire il lavoro della segreteria (nei programmi commerciali sono venduti separatamente),
oppure riguardano funzioni che sono implementate nelle piattaforme didattiche.


**Risposta lunga:**

>     * firma in compresenza delle lezioni scolastiche;
E' già possibile basta configurare correttamente le cattedre.


>     * possibilità di inserimento di allegati e video nella lezione, compiti e annotazioni;
I compiti si  possono inserire nell'agenda ed è possibile inserire link a materiali esterni. Aggiungere altro non ritengo che sia vantaggioso, perché è molto più efficace usare una apposita piattaforma didattica su cui inserire tutto.
 
>     * inserimento del campo _visibile al solo docente_ nelle lezioni e nei compiti, per inserire note personali della lezione;
E' già possibile inserire dei commenti visibili al solo docente (osservazioni personali)

>     * aggiunta della funzione _nota didattica_ e _altro tipo di nota,_ oltre alla nota disciplinare, per evitare confusione;
E' già esistente l'ANNOTAZIONE.
 
>     * aggiunta della funzione calendario, dove si inseriscono eventi e attività di classe e personali;
Esiste già. Gli eventi sono inseriti dallo staff. Compiti e verifiche dal docente.
  
>     * aggiunta della funzione _condivisione materiale_, in cui il genitore o studente può inviare al docente il materiale didattico o disciplinare (viceversa, il docente può inviarlo al genitore o studente;
Vale quanto detto prima: meglio usare piattaforma didattica, molto più adatta a questo tipo di comunicazioni.
 
>     * nello scrutinio, aggiungere la funzione di importazione da proposta a scrutinio, funzione dei verbali e variabili;
I verbali sono già creati.

>     * aggiungere la funzione della modulistica;
I moduli sono gestiti, seppure con qualche limitazione.
 
>     * aggiungere la funzione di adozione libri;
>     * aggiungere la funzione dei pagamenti;
Riguardano funzioni del gestionale della segreteria

>     * aggiungere la funzione questionari;
Meglio usare una piattaforma didattica.
 
>     * aggiunta della funzione dell'amministrazione digitale e dell'albo online; 
>     * aggiunta della funzione della costruzione del sito web scolastico;
Riguardano funzioni del gestionale della segreteria o simili. Per il sito esistono ottimi CMS open source.
 
>     * aggiunta della funzione di posta e messaggi (sms);
Esiste già la notifica via email o via Telegram. Gli SMS sono un mezzo oramai in disuso (e spesso anche a pagamento), meglio interfacciarsi con le app degli smartphone
 
>     * aggiungere la funzione comunicazione tramite posta, personalizzabili, con funzione di inviarli a chi vuoi (es. comunicazione uno ad alunni, comunicazione due a genitori e docenti);
Sono del parere che il registro debba gestire solo le comunicazioni istituzionali: per il resto è meglio utilizzare altri sistemi (la piattaforma didattica è la più indicataa in tal senso).

>     * aggiunta della funzione esami di stato e di maturità;
L'esame di Stato è gestito dall'applicativo del MIUR (per la scuola superiore).

>     * aggiunta della funzione SIDI;
Riguarda le funzioni del gestionale di segreteria
 
>     * modifica della carta intestata (ognuno può modificare la carta intestata del proprio istituto, decidendo di togliere _istituto d'istruzione superiore_, che di solito appare nella consegna delle credenziali);
Si può già fare
 
>     * aggiunta della funzione di creazione dei documenti (come circolari) tramite il registro, con carta intestata, come per esempio Word, Open Office o Libre Office. 
Questo è previsto e prima o poi verrà implementato.

>     * possibilità di inserire la tipologia di lezione (es: _lezione ordinaria_, _lezione laboratoriale_ etc.)
>     * possibilità di inserire direttamente i compiti alla firma della lezione;
>     * nella funzione _documenti_, aggiungere la funzione _documenti per la classe, documenti per il docente, documenti per la classe e materia, documenti per alunno, documenti personali_;
>     * aggiunta della funzione _elaborato_, dove si inseriscono gli elaborati degli esami, tesine o variabili;
>     * aggiunta della funzione di alternanza alla scuola e al lavoro;
>     * nello scrutinio, aggiungere la funzione di importazione da proposta a scrutinio
>     * aggiunta della funzione delle firme digitali;
Queste sono funzioni da valutare per un eventuale inserimento.

Infine, ci tengo a rimarcare una cosa fondamentale: **_le idee non mancano, mancano le persone che le realizzino._**
Il valore aggiunto dell'open source è che chiunque (con competenze informatiche) può aggiungere migliorie al codice e condividerle.

**Reply by giadaschiavone** [reply]
Ciao Antonello, in questi giorni ci ho pensato e ho deciso che magari potrei riuscire ad aiutare. Almeno provare a fare il mio piccolo con le conoscenze che ho.
Fammi sapere, grazie!

**Reply by trinko** [reply]
> Ciao Antonello, in questi giorni ci ho pensato e ho deciso che magari potrei riuscire ad aiutare. Almeno provare a fare il mio piccolo con le conoscenze che ho. Fammi sapere, grazie!

Ti ringrazio per la disponibilità. Tieni presente che:
- per effettuare modifiche sul codice è necessario conoscere il framework [Symfony](https://symfony.com/) e per l'uso del database anche [Doctrine](http://www.doctrine-project.org/)
- per l'interfaccia grafica, si usano i template [Twig](https://twig.symfony.com/), con [Jquery](https://jquery.com/) e [Bootstrap](https://getbootstrap.com/)

Per lo schema del database e per come sono strutturate le procedure non c'è purtroppo documentazione, non ho tempo di farla.

Insomma, lo so che è complicato...
